### PR TITLE
fix: Remove notifications background in AppLayout when there is no header nor breadcrumbs

### PR DESCRIPTION
**Pull request by abdhalees**
### Description

In AppLayout when there is no header or breadcrumbs, Notifications has a dark background even though there is no other elements in the page that needs this background. 
[see current behaviour](https://d21d5uik3ws71m.cloudfront.net/components/b9e7a91737590f92b5488c5beca5363f9f57b495/index.html#/light/app-layout/with-notifications) 

new behaviour: -will add it after preview is deployed-

### How has this been tested?

Already covered by screenshot tests in [with-notifications.page.tsx](https://github.com/cloudscape-design/components/blob/main/pages/app-layout/with-notifications.page.tsx), [with-breadcrumbs-notifications.page.tsx](https://github.com/cloudscape-design/components/blob/main/pages/app-layout/with-breadcrumbs-notifications.page.tsx) and a few others that covers all permutations.

the page with notifications only will break our screenshot tests but that is expected, the rest should stay the same

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/visual-refresh/context.tsx [hunk 1]
```diff
@@ -491,6 +493,23 @@ export const AppLayoutInternalsProvider = React.forwardRef(
...
+     * The disableContentHeaderOverlap property is absolute and will always disable the overlap
+     * if it is set to true. If there is no contentHeader then the overlap should be disabled
+     * unless there is a dynamicOverlapHeight. The dynamicOverlapHeight property is set by a
+     * component in the content slot that needs to manually control the overlap height. Components
+     * such as the Table (full page variant), Wizard, ContentLayout use this property and will
+     * retain the overlap even if there is nothing rendered in the contentHeader slot.
+     */
+    let isOverlapDisabled = false;
+
+    if (props.disableContentHeaderOverlap) {
```
[the remark this answers is not in the document]
Comment (abdhalees) [reply src/app-layout/visual-refresh/context.tsx:507]: blind copy pasting 🤦🏻‍♂️

File: src/app-layout/visual-refresh/context.tsx [hunk 2]
```diff
@@ -545,6 +564,8 @@ export const AppLayoutInternalsProvider = React.forwardRef(
           toolsRefs,
           navigationRefs,
           loseToolsFocus,
+          isOverlapDisabled,
+          hasBreadcrumbs: !!props.breadcrumbs,
```
Comment (DaemonCahill) [inline src/app-layout/visual-refresh/context.tsx:568]: I don't think we need this. You can just use `breadcrumbs` directly which is already in the the context.

File: src/app-layout/visual-refresh/background.tsx [hunk 3]
```diff
@@ -10,8 +10,11 @@ import styles from './styles.css.js';
  * that the design tokens used are overridden with the appropriate values.
  */
 export default function Background() {
-  const { hasNotificationsContent, hasStickyBackground, stickyNotifications } = useAppLayoutInternals();
-
+  const { hasNotificationsContent, hasStickyBackground, stickyNotifications, hasBreadcrumbs, isOverlapDisabled } =
+    useAppLayoutInternals();
+  if (hasNotificationsContent && !hasBreadcrumbs && isOverlapDisabled) {
```
Comment (DaemonCahill) [inline src/app-layout/visual-refresh/background.tsx:15]: Per my other comment we can just check `!breadcrumbs` and save ourselves the additional property. I think I already have code doing that in other places.

**Comment by DaemonCahill** [comment]
A few minor comments, otherwise looks good!

**Review by DaemonCahill — APPROVED** [review]

**Pull request closed without merging**